File: algorithmic-toolbox/week2/assignment/java/LCM.java
import java.util.*;
import java.math.BigInteger;

public class LCM {

    private static long getGCD(long a, long b){
        if (b == 0) return a;

        return getGCD(b, a % b);
    }

    private static BigInteger getLCM(long a, long b){
        BigInteger m = BigInteger.valueOf(a);
        BigInteger n = BigInteger.valueOf(b);
        BigInteger gcd = BigInteger.valueOf(getGCD(a, b));
        return m.multiply(n).divide(gcd);
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long a = scanner.nextLong();
        long b = scanner.nextLong();

        BigInteger c = getLCM(a, b);
        System.out.println(c);
    }
}

File: algorithmic-toolbox/week2/assignment/go/lcm.go
package main

import "fmt"

func lcm(a, b int64) int64 {
	gcdnum := gcd(a, b)
	return (a * b) / gcdnum
}

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	remainder := a % b
	return gcd(b, remainder)
}

func main() {
	fmt.Printf("%d\n", lcm(1827116622, 251772294))
}
